### DISCUSSION TITLE
But what do you do about train programming?

### DISCUSSION THREAD

**Discussion by steve-anthony-lee** [Opinionated-Blueprints/10-Books-Full-of-Rails#82] [General]
I'm still a little shakey on how to program the stations (I was used to the LTN system before). But what should I be doing to the trains themselves? How do I make sure they're on the right network? Do I add conditions manually? Do I mess with interrupts?

**Comment by Bocian-1** [comment]
https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/wiki

**Reply by steve-anthony-lee** [reply]
Yeah I read that but it doesn't mention what to do about the trains themselves, unless I'm blind

**Reply by Bocian-1** [reply]
Nothing beyond picking the group. Trains only go to stations in the same groups. You can be fancy with it if you want, but it's mainly meant to separate train configurations. I like to pick the numbered signals for the length of the train. I run 2-4 trains mostly but have a few 1-1 mixed in, so I'd use the [Signal 6] for 2-4 and [Signal 2] for 1-1. For fluid I pick the same, but uncommon quality.

**Reply by steve-anthony-lee** [reply]
Gotcha and how do you properly set a train to a group? Do you name it [group #] and that's it? Could I call a train "[1] Copper" or would that fuck things up?

**Reply by Bocian-1** [reply]
You just pick a train and pick an icon from this dialogue to represent the group.
![image](https://github.com/user-attachments/assets/f11621a2-837a-46a3-acc9-23cbef7df336)

**Reply by steve-anthony-lee** [reply]
Ooooh basically use the blueprint to spawn a train. Got it. That's where I fucked up. So basic order of operation is (for example for a mining station and smelter drop off)
1. Place provider station blueprint near mining area, configure it for group 1 etc..
2. Place requester station blueprint near smelters, group 1
3. Place train blueprint wherever, also group 1
4. Place fueling station somewhere like on a stacker, group 1. 

And that's basically it?

**Reply by Bocian-1** [reply]
Yes, don't forget to fill out the rest of the popup for providers/requesters though 👍

**Reply by steve-anthony-lee** [reply]
The rest.. like the wagon size, stack size and other stuff?

**Reply by Bocian-1** [reply]
yes

**Reply by steve-anthony-lee** [reply]
Sweet, thanks!

**Reply by steve-anthony-lee** [reply]
Hmm, well I had a requester and provider station down already. I redid the train like you suggested but now it's saying can't find path? Is there a debug menu?

**Reply by Bocian-1** [reply]
I don't understand what the problem is anymore.
Maybe one of my saves will give a better demonstration
My editor/test save:
[SA RAIL.zip](https://github.com/user-attachments/files/17910100/SA.RAIL.zip)
My freeplay save:
[Space Age.zip](https://github.com/user-attachments/files/17910103/Space.Age.zip)

**Reply by steve-anthony-lee** [reply]
No I the I figured it out. My intersections are all fucked up. It literally couldn't find a path because I failed to notice the intersections weren't built right. So your code worked!
